src/seq_mv/csr_spgemm_device.h

```diff
@@ -14,38 +14,256 @@
 
 #define COHEN_USE_SHMEM 0
 
-/* these are under the assumptions made in spgemm on block sizes: only use in spmm routines
- * where we assume CUDA block is 3D and blockDim.x * blockDim.y = WARP_SIZE
- */
+static const char HYPRE_SPGEMM_HASH_TYPE = 'D';
+
+/* default settings associated with bin 5 */
+#if defined(HYPRE_USING_CUDA)
+#define HYPRE_SPGEMM_NUMER_HASH_SIZE 256
+#endif
+#if defined(HYPRE_USING_HIP)
+#define HYPRE_SPGEMM_NUMER_HASH_SIZE 128
+#endif
+#define HYPRE_SPGEMM_SYMBL_HASH_SIZE 512
+#define HYPRE_SPGEMM_BASE_GROUP_SIZE 32
+/* unroll factor in the kernels */
+#if defined(HYPRE_USING_CUDA)
+#define HYPRE_SPGEMM_NUMER_UNROLL 256
+#define HYPRE_SPGEMM_SYMBL_UNROLL 512
+#elif defined(HYPRE_USING_HIP)
+#define HYPRE_SPGEMM_NUMER_UNROLL 256
+#define HYPRE_SPGEMM_SYMBL_UNROLL 512
+#endif
+
+//#define HYPRE_SPGEMM_TIMING
+//#define HYPRE_SPGEMM_PRINTF
+//#define HYPRE_SPGEMM_NVTX
+
+/* ----------------------------------------------------------------------------------------------- *
+ * these are under the assumptions made in spgemm on block sizes: only use in csr_spgemm routines
+ * where we assume CUDA block is 3D and blockDim.x * blockDim.y = GROUP_SIZE
+ *------------------------------------------------------------------------------------------------ */
+
+/* the number of threads in the block */
 static __device__ __forceinline__
 hypre_int get_block_size()
 {
-   //return (blockDim.x * blockDim.y * blockDim.z);           // in general cases
-   return (HYPRE_WARP_SIZE * blockDim.z);                     // if blockDim.x * blockDim.y = WARP_SIZE
+   return (blockDim.x * blockDim.y * blockDim.z);
 }
 
+/* the thread id in the block */
 static __device__ __forceinline__
 hypre_int get_thread_id()
```

Comment:
Same here with `hypre_cuda_get_thread_id`. I guess the idea is to replace the utilities defined here with the `hypre_cuda_` ones defined at `device_utils.h` when possible

Reply:
@liruipeng, I'm also curious why you don't use the existing functionality from `device_utils.h` here for things like getting the thread/warp/lane IDs etc.